translates: tutorial (salvaged) (#73)
reactjs/nl.reactjs.org: content/tutorial/tutorial.md

  - "docs/tutorial-zh-CN.html"
---

> Deze tutorial vereist geen voorbestaande React kennis.

Author: ```suggestion
Deze tutorial veronderstelt geen voorkennis van React.
```

## Voordat We Beginnen {#before-we-start-the-tutorial}

We zullen een klein spelletje maken in deze tutorial. **Misschien ben je geneigd om dit over te slaan aangezien je niet geïnteresseerd bent in het ontwikkelen van spelletjes -- maar geef het een kans.** De technieken die je zal leren in deze tutorial zijn fundamenteel om React applicaties te bouwen, en als je die onder de knie hebt geeft dat je een diep begrip van React. 

>Tip
>

> >Deze tutorial is bedoeld voor zij die het liefst **al doende leren**. Indien je liever concepten stap voor stap leert, neem dan een kijkje in de [stapsgewijze documentatie](/docs/hello-world.html).

Author: ```suggestion
>Deze tutorial ontworpen voor mensen die het liefst **al doende leren**. Als je concepten liever vanaf de basis leert, neem dan een kijkje in de [stapsgewijze documentatie](/docs/hello-world.html).
```

De tutorial is opgedeeld in verschillende onderdelen:

> * [Setup Voor De Tutorial](#setup-for-the-tutorial) geeft je **een beginpunt** om de tutorial te volgen.

Author: ```suggestion
* [Setup voor de Tutorial](#setup-for-the-tutorial) geeft je **een beginpunt** om de tutorial te volgen.
```

> * [Overzicht](#overview) leert je de **fundamentele concepten** van React: componenten (components), eigenschappen (props), en staat (state).

Author: ```suggestion
* [Overzicht](#overview) leert je de **fundamentele concepten** van React: componenten, props, en state.
```

> * [Het Spel Afwerken](#completing-the-game) leert je de **meest voorkomende technieken** in React ontwikkeling.

Author: ```suggestion
* [Het Spel Afwerken](#completing-the-game) leert je de **meest voorkomende technieken** bij het ontwikkelen in React.
```

> * [Tijdreizen Toevoegen](#adding-time-travel) zal je **een dieper inzicht** geven in de unieke voordelen van React.

Author: ```suggestion
* [Tijdreizen Toevoegen](#adding-time-travel) zal je **een dieper inzicht** geven in de unieke sterke punten van React.
```

> Het is niet noodzakelijk om alle onderdelen in één keer te voltooien. Probeer zo ver mogelijk te komen -- zelfs al is het een of twee onderdelen.

Author: ```suggestion
Je hoeft niet alle onderdelen in één keer af te maken. Probeer zo ver mogelijk te komen -- zelfs al is het één of twee onderdelen.
```

### Wat Gaan We Maken? {#what-are-we-building}

> In deze tutorial, zullen we tonen hoe je een interactief tic-tac-toe (boter-kaas-eieren) spel kan maken met React.

Author: ```suggestion
In deze tutorial, zullen we laten zien hoe je een interactief tic-tac-toe (boter-kaas-eieren) spel kan maken met React.
```

> Je kan een kijkje nemen naar wat we zullen maken: **[Afgewerkt Resultaat](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**. Als de code nog geen steek houdt of je syntax nog niet kent, is dat geen probleem. Het doel van deze tutorial is je React en de syntax aan te leren.

Author: ```suggestion
Je kunt hier bekijken wat we zullen gaan bouwen: **[Eindresultaat](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**. Maak je geen zorgen als de code nog niet logisch lijkt of als je nog onbekend bent met de syntax van de code. Het doel van deze tutorial is om je te helpen React en zijn syntax te begrijpen.
```

> We raden aan dat een kijkje neemt naar het tic-tac-toe spel voordat je verdergaat met de tutorial. Een van de eigenschappen dat je zal opmerken is dat er een genummerde lijst is aan de rechterkant van het spel bord. Deze lijst geeft je een geschiedenis van alle zetten die in het spel zijn voorgekomen, en deze wordt geupdate tijdens het spel.

Author: ```suggestion
We raden aan dat je het tic-tac-toe spel bekijkt voordat je verdergaat met de tutorial. Eén van de functionaliteiten die je zal opvallen is een genummerde lijst aan de rechterkant van het spelbord. De lijst geeft je de geschiedenis van alle zetten die in het spel zijn gebeurd, en deze wordt geüpdate tijdens het spel.
```

> Eens je vertrouwd bent met het tic-tac-toe spel, kan je het afsluiten. We zullen vanuit een simpelere template starten voor deze tutorial.  Onze volgende stap is jouw setup in te stellen zodat je het spel kan beginnen te bouwen.

Author: ```suggestion
Je kunt het tic-tac-toe spel afsluiten zodra je er vertrouwd mee bent. We zullen deze tutorial beginnen vanuit een eenvoudiger template.  Onze volgende stap is je voor te bereiden om te kunnen starten met het bouwen van het spel.
```

> ### Vereisten {#prerequisites}

Author: ```suggestion
### Eerste Vereisten {#prerequisites}
```

> We gaan ervanuit dat je enigzins vertrouwd bent met HTML en Javascript, maar het is mogelijk om deze tutorial te volgen, zelfs als je van een andere programmeer taal komt. We gaan er ook vanuit dat je vertrouwd bent met programmeer concepten zoals functies (functions), objecten (objects), arrays en in mindere mate met klasses (classes).

Author: ```suggestion
We gaan ervan uit dat je enigszins vertrouwd bent met HTML en Javascript, maar je zou dit moeten kunnen volgen, ookal kom je vanuit een andere programmeertaal. We gaan er ook vanuit dat je vertrouwd bent met programmeer concepten zoals functies (functions), objecten (objects), arrays en in mindere mate met klassen (classes).
```

> Als je Javascript wil herhalen, raden we [deze handleiding](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript). We gebruiken ook sommige functies van ES6 -- een recente versie van Javscript. In deze tutorial zullen we [arrow functies](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), [classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes), [`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let), en [`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) statements gebruiken. Je kan de [Babel REPL](babel://es5-syntax-example) gebruiken om de controleren hoe ES6 code compileert.

Author: ```suggestion
Als je Javascript wilt herhalen, raden we aan [deze handleiding](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript) te lezen. Houd er rekening mee dat we ook enkele functies van ES6 gebruiken -- een recente versie van Javscript. In deze tutorial zullen we [arrow functies](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), [classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes), [`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let), en [`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) statements gebruiken. Je kan de [Babel REPL](babel://es5-syntax-example) gebruiken om de controleren hoe ES6 code compileert.
```

## Setup voor de Tutorial {#setup-for-the-tutorial}

> There are two ways to complete this tutorial: you can either write the code in your browser, or you can set up a local development environment on your computer.

Author: Er zijn twee manieren om deze tutorial te voltooien: je kunt de code in je browser schrijven of een lokale ontwikkelomgeving opzetten op je computer.

}
```

Voor:

![React Devtools](../images/tutorial/tictac-empty.png)

Na: Je zal normaal een nummer in elk vierant van de gerenderde output zien.

![React Devtools](../images/tutorial/tictac-numbers.png)

**[De volledige code tot zover](https://codepen.io/gaearon/pen/aWWQOG?editors=0010)**

Gefeliciteerd, je het zojuist een prop van de parent Board doorgegeven aan het child Square component door gegeven. Props doorgeven is hoe informatie vloeit in React applicaties, van parents naar children.

> ### Interactieve Componenten maken {#making-an-interactive-component}

Author: ```suggestion
### Een Interactieve Component maken {#making-an-interactive-component}
```

    // the rest has not changed
```

Als we een stap in de spel geschiedenis aanklikken, zou het tic-tac-toe bord onmiddelijk updaten om het bord te tonen zoals het eruit ziet net nadat de zet gemaakt is.

**[De volledige code tot zover](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**

### Tot Slot {#wrapping-up}

Gefeliciteerd! Je hebt zojuist een tic-tac-toe game gemaakt dat:

* Je tic-tac-toe laat spelen
* Aantoont wanneer een speler gewonnen heeft
* De geschiedenis van alle zetten opslaat 
* Spelers toe laat om de geschiedenis van alle zetten en versies van het bord te bekijken

Mooi zo! We hopen dat je nu een goed overzicht hebt gekregen van hoe React werkt.

Bekijk het eind resultaat: **[Eind Resultaat](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**.

Als je nog extra tijd hebt of je React vaardigheden wat meer wil oefenen, zijn hieronder nog wat ideëen voor verbeteringen die je kan aanbrengen op het tic-tac-toe spel, in opgaande moeilijkheid:

1. Toon de locatie van elke zet in het formaat (kolom, rij) in de geschiedenislijst van de zetten.

> 2. Zet het huidig geselecteerde item in de geschiedenislijst in het vet.

Author: ```suggestion
2. Maak het huidig geselecteerde item in de geschiedenislijst vet gedrukt.
```

> 3. Herschrijf het Board component, gebruik makend van twee loops om de vierkanten te maken i.p.v. deze te hardcoden.

Author: ```suggestion
3. Herschrijf het Board component met twee loops om de vakjes te maken in plaats van deze te hardcoden.
```

> 4. Voeg een wisselknop toe die toelaat om de geschiedenislijst in opgaande of afnemede volgorde te schikken.

Author: ```suggestion
4. Voeg een wisselknop toe die toelaat om de geschiedenislijst in oplopende of aflopende volgorde te sorteren.
```

> 5. Wanneer iemand wint, duid de vakken aan die de overwinning hebben gebracht.

Author: ```suggestion
5. Wanneer iemand wint, markeer de vakken die voor de overwinning zorgden.
```

> 6. Wanneer er geen winnaar is, toon een boodschap die een gelijkspel vermeldt.

Author: Als er niemand wint, toon een boodschap dat het resultaat een gelijkspel is.
